llama.h

```diff
@@ -284,6 +313,9 @@ extern "C" {
     /// @details Randomly selects a token from the candidates based on their probabilities.
     LLAMA_API llama_token llama_sample_token(struct llama_context * ctx, llama_token_data_array * candidates);
 
+    /// @details Accepts the sampled token into the grammar, possibly transforming to a new token
+    LLAMA_API llama_token llama_grammar_accept_token(struct llama_context * ctx, struct llama_grammar * grammar, llama_token token);
+
```

Comment:
In what cases the token is being transformed?

Reply:
When only a prefix of the sampled token's chars match the grammar. For example, if the grammar is expecting "}" followed by, say, whitespace or a comma, and you sample "}{", you'd get back the "}" token. This can happen because this implementation only peeks one char ahead for the sampling penalties.

Reply:
> This can happen because this implementation only peeks one char ahead for the sampling penalties.

I see. So it wouldn't happen if the sampling penalties were applied optimally according to the grammar, but for now it's much easier to have the implementation peek at just one char, correct?

Reply:
Yes, correct.